src/ProviderPR/SFA.DAS.ProviderPR/Application/Queries/GetEASUserByEmail/GetEasUserByEmailHandler.cs

```diff
@@ -0,0 +1,40 @@
+﻿using MediatR;
+using SFA.DAS.SharedOuterApi.Configuration;
+using SFA.DAS.SharedOuterApi.InnerApi.Requests;
+using SFA.DAS.SharedOuterApi.InnerApi.Requests.EmployerAccounts;
+using SFA.DAS.SharedOuterApi.InnerApi.Responses;
+using SFA.DAS.SharedOuterApi.InnerApi.Responses.EmployerAccounts;
+using SFA.DAS.SharedOuterApi.Interfaces;
+
+namespace SFA.DAS.ProviderPR.Application.Queries.GetEasUserByEmail;
+public class GetEasUserByEmailQueryHandler(IAccountsApiClient<AccountsConfiguration> apiClient) : IRequestHandler<GetEasUserByEmailQuery, GetEasUserByEmailQueryResult>
+{
+
+    public async Task<GetEasUserByEmailQueryResult> Handle(GetEasUserByEmailQuery request, CancellationToken cancellationToken)
+    {
+        var result = await apiClient.Get<GetUserByEmailResponse>(new GetUserByEmailRequest(request.Email));
+
+        var userRef = result?.Ref;
+
+        if (userRef == null || userRef == Guid.Empty)
```

Comment:
I get defensive coding, but I rather like my code to state the expectations correctly. Hence I would not check for empty guid at all as it is not expected to be empty, by adding this check we are adding a doubt that this can come back as empty.

Reply:
I've removed the check for guid.empty